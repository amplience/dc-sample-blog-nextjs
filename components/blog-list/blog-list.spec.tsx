/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BlogList from './blog-list';
import BlogCard from '../blog-card/blog-card';
jest.mock('../blog-card/blog-card', () => () => <div className="blog-card-mock" />);

describe('BlogList', () => {
  let blogPostFixture;

  beforeEach(() => {
    blogPostFixture = {
      title: 'blog-post-test',
      date: '2019-08-13',
      description: 'blog-post-description',
      authors: [],
      image: {
        image: {
          id: 'image-id',
          name: 'image-name',
          endpoint: 'image-endpoint',
          defaultHost: 'image-default-host'
        },
        altText: ''
      },
      tags: ['test-tag'],
      readTime: 15,
      content: []
    };
  });

  test('renders blog list with a single blog', async () => {
    const props = {
      blogPosts: [blogPostFixture]
    };
    const component = await renderer.create(<BlogList {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders blog list with ten blog posts', async () => {
    const props = {
      blogPosts: Array(10).fill(blogPostFixture)
    };

    const wrapper = shallow(<BlogList {...props} />);
    expect(wrapper.find(BlogCard).length).toEqual(10);
  });

  test('renders empty blog list when blog list is empty array', async () => {
    const props = {
      blogPosts: []
    };
    const component = await renderer.create(<BlogList {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders empty blog list when blog list is undefined', async () => {
    const props = {
      blogPosts: undefined
    };
    const component = await renderer.create(<BlogList {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
