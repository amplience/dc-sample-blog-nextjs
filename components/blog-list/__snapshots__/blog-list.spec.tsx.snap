// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList renders blog list with a single blog 1`] = `
<section
  className="jsx-2721295117"
>
  <div
    className="blog-card-mock"
  />
</section>
`;

exports[`BlogList renders empty blog list when blog list is empty array 1`] = `
<section
  className="jsx-2721295117"
>
  
</section>
`;

exports[`BlogList renders empty blog list when blog list is undefined 1`] = `
<section
  className="jsx-2721295117"
>
  
</section>
`;
